core/web/resolver/auth.go

```diff
@@ -3,16 +3,52 @@ package resolver
 import (
 	"context"
 
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/sessions"
 	"github.com/smartcontractkit/chainlink/core/web/auth"
 )
 
-// Authenticates the user from the session cookie.
+// Authenticates the user from the session cookie, presence of user inherently provides 'view' access.
 func authenticateUser(ctx context.Context) error {
-	_, ok := auth.GetGQLAuthenticatedSession(ctx)
+	if _, ok := auth.GetGQLAuthenticatedSession(ctx); !ok {
+		return unauthorizedError{}
+	}
+	return nil
+}
+
+// Authenticates the user from the session cookie and asserts at least 'run' role.
+func authenticateUserCanEditMinimal(ctx context.Context) error {
+	session, ok := auth.GetGQLAuthenticatedSession(ctx)
 	if !ok {
 		return unauthorizedError{}
 	}
+	if session.User.Role == sessions.UserRoleView {
+		return errors.Errorf("Not permitted with current role %s\n", session.User.Role)
+	}
+	return nil
+}
 
+// Authenticates the user from the session cookie and asserts at least 'edit' role.
+func authenticateUserCanEdit(ctx context.Context) error {
+	session, ok := auth.GetGQLAuthenticatedSession(ctx)
+	if !ok {
+		return unauthorizedError{}
+	}
+	if session.User.Role == sessions.UserRoleView || session.User.Role == sessions.UserRoleRun {
+		return errors.Errorf("Not permitted with current role %s\n", session.User.Role)
+	}
```

Comment:
A switch would be easy to read here:
```suggestion
	switch session.User.Role {
	case sessions.UserRoleView, sessions.UserRoleRun:
		return errors.Errorf("Not permitted with current role %s\n", session.User.Role)
	}
```

Reply:
Sure looks good. Updated